### ISSUE TITLE
Offering to help maintain Plumble

### ISSUE THREAD

**Issue by skorokithakis** [acomminos/Plumble#202]
I've noticed that activity on this project has slowed, and I was wondering if you needed some help. Would you consider adding the project to [Code Shelter](https://www.codeshelter.co/)? It's an open community of FOSS maintainers (please feel free to join!) that aims to help FOSS projects out (fixing bugs, merging PRs, releasing, etc), and I'm a member and can help.

You can [read more details about it](https://www.codeshelter.co/faq/) or [add your project](https://www.codeshelter.co/projects/add/) to it. There are also people available to help on the Code Shelter [chat channel](https://codeshelter.zulipchat.com/).

Thanks!

**Comment by poVoq** [comment]
Yeah that would be cool.

I think once Grumble includes a configuration option it could together with the awesome new mumble-web HTML5 client really result in a small revival of Mumble.

**Comment by quite** [comment]
It would be really cool! But is @acomminos interested?

The awesome new mumble-web HTML5 client also sounds cool! Where is it being developed?

**Comment by iammeat** [comment]
I've looked through his activity, and it shows lots of private repo commits as well as commits to the official Chromium and Facebook repos.  It might be that he has a new job with one of the big companies and is no longer able/willing to work on Plumble.  My suggestion is that we create a new fork of the project and begin developing it that way, as I've not seen anything on here for a couple of years from @acomminos.

We can always merge those changes back in if activity starts back up again.

Thoughts?

**Comment by quite** [comment]
Sounds like a good idea. And I got similar thoughts from around the project owner's profile.

**Comment by Akito13** [comment]
As mentioned in https://github.com/acomminos/Plumble/issues/197#issuecomment-456389142 I forked the project and tried to start working on it, but it ended up in a gradle-caused dependency hell due to the "old" (apparently a couple of years seems ancient from gradle's perspective) packages, etc. Due to that, I didn't even get it to work, in the first place. My friend, who is a professional Java programmer since several years now, took a look at it and not even he could figure out how to fix it. He basically told me, that all this has to be rewritten from scratch or implemented manually, without gradle.
So, it would be already a big help and a good progress enabler, if someone could fix the gradle hell in this project. Currently, I am pretty busy myself, but if the gradle hell is fixed I might start working on my fork from time to time. Of course it would be nice, if users would contribute, if they can.

Until now, I had a private fork on my own server, but I just forked it onto my public Github profile, making it available here https://github.com/Akito13/Plumble. I can easily take care of PR's and test/review them, even if I can't invest a lot of time in primarily developing the project myself.

**Comment by iammeat** [comment]
I managed to get it working on the latest gradle through lots of messing about, so it is possible.  If you like, I'll grab it from your fork and make the changes then submit a PR after the weekend.

**Comment by Akito13** [comment]
> 
> 
> I managed to get it working on the latest gradle through lots of messing about, so it is possible. If you like, I'll grab it from your fork and make the changes then submit a PR after the weekend.

That would be great. I will be happy when people like you and others alike will join and at least open up a small flame of life for this project. As mentioned, I can't guaranteee that I can invest a lot of time into the development myself, but I will gladly take care of PRs, Issues, etc. So this is at least a good beginning. 😃 

@Damianuscz @Mikaela @deluxghost @rock2rap

**Comment by iammeat** [comment]
Updated my last comment with some stuff :)

**Comment by Akito13** [comment]
> Make sure you also fork the Jumble repo and update the submodules to use yours instead of the one here.

I will do such stuff, when the first PRs come in, as then I will be motivated to merge some changes from my private repository, where I already started to explain this fork, upgraded submodules, changed the name of the fork, so it is distinguishable as a fork, etc.

**Comment by iammeat** [comment]
Okay.  Good stuff.  I'm on board to get the ball rolling forward on this.

**Comment by iammeat** [comment]
I will post future comments and discussions on your fork.

**Comment by skorokithakis** [comment]
Does anyone know if we can contact @acomminos so we can get commit privileges to the official repository? It would be great if we could release the app as well without having to fork everything.

**Comment by iammeat** [comment]
Several people have tried to contact him over the last couple of years, including me, but no replies have been forthcoming.  Worth another try I guess.

**Comment by skorokithakis** [comment]
I've sent him a friend request on Facebook, hopefully that works.

**Comment by Akito13** [comment]
> 
> 
> Does anyone know if we can contact @acomminos so we can get commit privileges to the official repository? It would be great if we could release the app as well without having to fork everything.

1. Commit privileges is not enough, if the admin is missing.
2. What is wrong with forks?

**Comment by iammeat** [comment]
I see from his official site that he now works at Facebook.

**Comment by Akito13** [comment]
> 
> 
> I see from his official site that he now works at Facebook.

Considering this, maybe we should consciously decoupling from this person.

**Comment by skorokithakis** [comment]
> Commit privileges is not enough, if the admin is missing.

If he adds the project to Code Shelter, we can have some admin privileges as well.

> What is wrong with forks?

**Comment by iammeat** [comment]
It's sad to leave a project such as this languishing as it is a solid piece of software.

**Comment by iammeat** [comment]
It's probably better to fork this project and then if he comes back to active status on it we can merge the fork changes back into the master.  Otherwise we're at his mercy waiting for a reply and I've waited 2 years already to get involved with this.

**Comment by Akito13** [comment]
> 
> If he adds the project to Code Shelter, we can have some admin privileges as well.



Maybe from your perspective Code Shelter seems like a worthwhile plus, but I don't think everyone agrees.

> Nothing, it would just be better if we could continue improving the original codebase.

Forking the project replicates the original codebase...
I think it would be better to leave the "original" project as it is, as it originally was from the mentioned admin. Now, as it will be maintained from an entirely different team, I think it is more than appropriate to fork this project. It fits and makes sense.

**Comment by quite** [comment]
Starting off getting some work done in @akito13's fork sounded like a good idea to me. Given that I think the project owner in unlikely to show up soon...

**Comment by iammeat** [comment]
@quite I agree.  I'm ready to start fixing the gradle issues right away, and that's the first step to getting it built with the latest release of AS.

**Comment by skorokithakis** [comment]
> It's probably better to fork this project and then if he comes back to active status on it we can merge the fork changes back into the master.

Yeah, agreed.

> Maybe from your perspective Code Shelter seems like a worthwhile plus, but I don't think everyone agrees.

What's the disagreement?

**Comment by iammeat** [comment]
There are also fixes needed for the way it builds the NDK libraries, so I'll take that into account when I make the changes.

**Comment by Akito13** [comment]
@iammeat @quite FYI, I started a first issue on an idea I had. Whoever is interested, should join the conversation. 😃 
https://github.com/Akito13/Plumble/issues/1

**Comment by iammeat** [comment]
I put my suggestions.

**Comment by deluxghost** [comment]
thanks for mention me. I'm really happy as you nice guys and open-source world make dream come ture.
I'm not a java programmer, but I'd like to join testing when needed

**Comment by deluxghost** [comment]
looks like there are some forks were trying to build with newer gradle
that might be helpful

**Comment by Akito13** [comment]
@deluxghost Thanks for the tip, we will consider taking these changes into account.
To keep things organized, we should keep further issues in https://github.com/Akito13/plumbly.
